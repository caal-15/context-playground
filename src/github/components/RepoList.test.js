import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, configure  } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import RepoList from './RepoList';

configure({ adapter: new Adapter() });

const defaultProps = {
  repos: []
}, RepoListLayout = (props) => <RepoList {...defaultProps} {...props} />;

describe('Render RepoListLayout', () => {
  it('render correctly RepoListLayout component', () => {
    const RepoListLayoutComponent = renderer.create(<RepoListLayout />).toJSON();
    expect(RepoListLayoutComponent).toMatchSnapshot();
  });

  it('render correctly required prop repos', () => {
    const props = {
      repos: [
        {
          archived: false,
          forks: 0,
          created_at: '2016-06-04T17:57:07Z',
          name: 'test1'
        },
        {
          archived: false,
          forks: 0,
          created_at: '2016-06-04T17:57:07Z',
          name: 'test2'
        }
      ]
    },
      RepoListLayoutComponent = mount(<RepoListLayout {...props} />);
    expect(RepoListLayoutComponent.find('.md-cell--6').length).toEqual(2);
});

});