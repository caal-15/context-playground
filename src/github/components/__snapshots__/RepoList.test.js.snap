// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render RepoListLayout render correctly RepoListLayout component 1`] = `
<div
  className="md-grid"
/>
`;

exports[`Render SidebarLayout render correctly SidebarLayout component 1`] = `
<div
  className="md-grid"
/>
`;
